test_fitness.py

```diff
@@ -0,0 +1,134 @@
+# Imports
+import csv
+import numpy as np
+import argparse
+import random
+import math
+
+# Functions
+
+def read_data(sections, genes, observed):
+    # Read in data from generationDNA.csv and put it into the observed list
+    with open('generationDNA.csv') as dna:
+        csv_read = csv.reader(dna, delimiter = ',')
+        individual = 0
+        for i, row in enumerate(csv_read):
+            if( i > 8):
+                if( (i-9)%sections == 0 ):
+                    j=0
+                elif( (i-9)%sections != 0 ):
+                    j=(i-9)%sections
+                for k in range(genes):
+                    observed[individual][j][k] = float(row[k])
+                if ( j == (i-9)%sections and (i-9)%sections !=0):
+                    individual = individual + 1
+                if ( sections == 1):
+                    individual = individual + 1
+    dna.close()
+            
+def solve_chi_squared(target, observed, sections, genes, chi2):
+    # Solve for Chi-squared scores
+    for i in range(0, g.population):
+        tempChi2 = 0
+        for j in range(sections):
+            for k in range(genes):
+                tempChi2 = tempChi2 + abs(((observed[i][j][k]-target[j][k])**2)/target[j][k])
+        chi2.append(tempChi2)
+  
+def calc_error(fitness, error):
+    # Calculate simulated error for the test
+    for i in range(0, len(fitness)):
+        if (fitness[i] == 0):
+            error.append(0.0)
+        else:
+            error.append(0.5/math.sqrt(41.0))
+            tempFit = fitness[i]
+            tempFit = random.gauss(fitness[i], 0.5/math.sqrt(1.0))
+            while(tempFit < 0):
+                tempFit = random.gauss(fitness[i], 0.5/math.sqrt(1.0))
+            fitness[i] = tempFit
+        
+def write_fitness(fitness, error):
+    # Write fintessScore csv with data for the generation
+    with open('fitnessScores.csv', "r") as fs:
+        lines = fs.readlines()
+    fs.close()
+            
+    lines2=[]            
+    with open('fitnessScores.csv', "w") as f2:
+        for x in range(0, len(fitness)+2):
+            if x <= 1:
+                lines2.append(str(lines[x]))
+            elif x>1:
+                lines2.append(str(fitness[x-2]) +"," + str(error[x-2]) + '\n')
+        f2.writelines(lines2)
+    f2.close
+    
+def write_chi(chi2):
+    # Write chiScore csv with data for the generation
+    with open('chiScores.csv', "r") as fc:
+        lines = fc.readlines()
+    fc.close()
+            
+    lines3=[]            
+    with open('chiScores.csv', "w") as f2:
+        for x in range(0, len(fitness)+2):
+            if x <= 1:
+                lines3.append(str(lines[x]))
+            elif x>1:
+                lines3.append(str(chi2[x-2]) + '\n')
+        f2.writelines(lines3)
+    f2.close
```

Comment:
```suggestion
def write_chi(chi2):
    # Write chiScore csv with data for the generation
    with open('chiScores.csv', "r") as fc:
        lines = fc.readlines()
            
    lines3 = []
    with open('chiScores.csv', "w") as f2:
        for x in range(len(fitness)+2):
            if x <= 1:
                lines3.append(str(lines[x]))
            elif x > 1:
                lines3.append(str(chi2[x-2]) + '\n')
        f2.writelines(lines3)
```
Since this code was duplicated, it also has the same bug.  Copying and pasting code is almost always wrong.